kong.conf.default

```diff
@@ -447,6 +447,16 @@
                                  # See the `lua_ssl_trusted_certificate`
                                  # setting to specify a certificate authority.
 
+#pg_semaphore_max = 0            # Sets the number of semaphore resources made
+                                 # available to the semaphore used to limit
+                                 # concurrent connections to the Postgres
+                                 # database. The default value of 0 removes the
+                                 # semaphore concurrency limiter.
+
+#pg_semaphore_timeout = 60       # Defines the timeout, in seconds, after which
+                                 # Postgres query semaphore resource acquisition
+                                 # attempts will fail.
```

Comment:
We should describe what happens when resource acquisition fails. Limiting the number of connections to Postgres isn't a hard problem because of implementation (e.g. semaphore which was suggested a while ago already, or native connection queuing in OpenResty 1.15.8.1), but because of the behavior to follow in unhappy cases. The problem isn't solved by connection queuing, but only "moved". Two options:

1. Failing hard, and rejecting the PostgreSQL query, which could result in an HTTP 503 error from the Gateway. Imho, most users would not be ok with such as behavior. An easily overwhelmed API Gateway seems to be antithetical with the sheer concept of an API Gateway.
2. Gracefully fail, and proceed with the query nonetheless to ensure resiliency. Which completely defeats the purpose for such a limit in the first place.

It is my understanding that users who wish for the second behavior can achieve it with `pg_semaphore_max = 0` (i.e. not providing a limit in the first place). Nonetheless, when a limit is provided, better be upfront about what the potential downsides of it are.

Reply:
I definitely do think we should detail this out very well. I'm not sure how far into detail to go here. The online docs seems like a better place to go into depth about various impacts and expected behaviors (as well as the current default ones). To me trying to explain it all in the reference config file would be too wordy and make this file unwieldy. That said this could go into a bit more detail about what would happen, although that occurs on a case by case basis, so the explanation here would be a bit vague.

To be super clear, when I talk about spelling things out in docs, I mean in a separate guide, not just the config file reference page at https://docs.konghq.com/1.1.x/configuration/

Reply:
Indeed. My comment here was only to add a brief note: "Failure to acquire a resource can result in a failure to proxy a request. Find out more in the xyz guide at ...". It is really important we at least hint at the downsides of this model in this file.

Reply:
I've updated the variable name per this suggestion. To me it makes the following `pg_semaphore_timeout` name a bit incongruous as it also doesn't hide an implementation detail? But maybe it's worth the detail here? :man_shrugging:

Reply:
Added in a bit more detail, with a hanging reference to the docs (once the docs PR is merged I will add a link here)